### PR TITLE
Added Option Pricing Exercises

### PR DESCRIPTION
**Pull request by jstac**

### CODE REVIEW

**Comment by jstac** [comment]
@HumphreyYang @mmcky , would one of you mind fixing the build errors and reviewing the PR?

**Comment by HumphreyYang** [comment]
> @HumphreyYang @mmcky , would one of you mind fixing the build errors and reviewing the PR?

I think it is just because the default values are not set. It should be fixed now.

**Comment by mmcky** [comment]
thanks @HumphreyYang

File: lectures/parallelization.md [hunk 1]
```diff
@@ -486,3 +486,113 @@ on the number of CPUs on your machine.)
...
+
+
+```{solution-start} parallel_ex2
+:class: dropdown
+```
+
+
+With $s_t := \ln S_t$, the price dynamics become
+
+$$ s_{t+1} = s_t + \mu + \exp(h_t) \xi_{t+1} $$
```
Comment (mmcky) [inline lectures/parallelization.md]:
```suggestion
$$
s_{t+1} = s_t + \mu + \exp(h_t) \xi_{t+1}
$$
```

File: lectures/scipy.md [hunk 2]
```diff
@@ -440,6 +440,127 @@ We leave you to investigate the [set of available routines](http://docs.scipy.or
 
 ## Exercises
 
+The first few execises concern pricing a European call option under the
+assumption of risk neutrality.  The price satisfies
+
+$$ P = \beta^n \mathbb E \max\{ S_n - K, 0 \} $$
```
Comment (mmcky) [inline lectures/scipy.md]:
```suggestion
$$
P = \beta^n \mathbb E \max\{ S_n - K, 0 \}
$$
```

File: lectures/scipy.md [hunk 3]
```diff
@@ -440,6 +440,127 @@ We leave you to investigate the [set of available routines](http://docs.scipy.or
...
+```{exercise}
+:label: sp_ex01
+
+Suppose that $S_n$ has the [log-normal](https://en.wikipedia.org/wiki/Log-normal_distribution) distribution with parameters $\mu$ and $\sigma$.  Let $f$ denote the density of this distribution.  Then
+
+$$ P = \beta^n \int_0^\infty \max\{x - K, 0\} f(x) dx $$
+
+Plot the function 
+
+$$g(x) = \beta^n  \max\{x - K, 0\} f(x)$$
```
Comment (mmcky) [inline lectures/scipy.md]:
```suggestion
$$
g(x) = \beta^n  \max\{x - K, 0\} f(x)
$$ 
```

File: lectures/scipy.md [hunk 4]
```diff
@@ -440,6 +440,127 @@ We leave you to investigate the [set of available routines](http://docs.scipy.or
...
+
+The price is the expectation of the payoff, discounted to current value.
+
+
+```{exercise}
+:label: sp_ex01
+
+Suppose that $S_n$ has the [log-normal](https://en.wikipedia.org/wiki/Log-normal_distribution) distribution with parameters $\mu$ and $\sigma$.  Let $f$ denote the density of this distribution.  Then
+
+$$ P = \beta^n \int_0^\infty \max\{x - K, 0\} f(x) dx $$
```
Comment (mmcky) [inline lectures/scipy.md]:
```suggestion
$$
P = \beta^n \int_0^\infty \max\{x - K, 0\} f(x) dx
$$
```

**Review by mmcky — COMMENTED** [review]
@jstac just really minor style suggestion re: `display math`

**Comment by HumphreyYang** [comment]
Hi @jstac and @mmcky,

Another style-related update: I replaced the `Hint:` with hint admonitions.

**Comment by jstac** [comment]
Thanks guys.  Nice catch @HumphreyYang .

@mmcky , 

- should exercises end with {exercise-end} or just backticks?  
- please merge when ready.

**Comment by mmcky** [comment]
> * should exercises end with {exercise-end} or just backticks?

hey @jstac there are two types. If there are any embedded directives (i.e. code or math (using ticks)) then the exercise will need to start with `exercise-start` and then end with `exercise-end`. however if the gated syntax is not used then it can be a standard directive that starts with `exercise` and ends with the standard back ticks to close the directive.

File: lectures/parallelization.md [hunk 5]
```diff
@@ -486,3 +486,113 @@ on the number of CPUs on your machine.)
...
+1. $\beta$ is a discount factor,
+2. $n$ is the expiry date,
+2. $K$ is the strike price and
+3. $\{S_t\}$ is the price of the underlying asset at each time $t$.
+
+Suppose that `n, β, K = 20, 0.99, 100`.
+
+Assume that the stock price obeys 
+
+$$ \ln \frac{S_{t+1}}{S_t} = \mu + \sigma_t \xi_{t+1} $$
```
Comment (mmcky) [inline lectures/parallelization.md]:
```suggestion
$$ 
\ln \frac{S_{t+1}}{S_t} = \mu + \sigma_t \xi_{t+1}
$$
```

File: lectures/parallelization.md [hunk 6]
```diff
@@ -486,3 +486,113 @@ on the number of CPUs on your machine.)
...
+
+In {doc}`our lecture on SciPy<scipy>`, we discussed pricing a call option in a
+setting where the underlying stock price had a simple and well-known
+distribution.
+
+Here we discuss a more realistic setting.
+
+We recall that the price of the option obeys 
+
+$$ P = \beta^n \mathbb E \max\{ S_n - K, 0 \} $$
```
Comment (mmcky) [inline lectures/parallelization.md]:
```suggestion
$$
P = \beta^n \mathbb E \max\{ S_n - K, 0 \}
$$
```

**Comment by mmcky** [comment]
@jstac are you OK with the commit suggestions listed above? They are all the same just adding a new line to mark the display math block

**Comment by jstac** [comment]
> @jstac are you OK with the commit suggestions listed above? They are all the same just adding a new line to mark the display math block

For sure, please go ahead @mmcky

File: lectures/scipy.md [hunk 7]
```diff
@@ -440,12 +440,144 @@ We leave you to investigate the [set of available routines](http://docs.scipy.or
 
 ## Exercises
 
+The first few execises concern pricing a European call option under the
```
Comment (mmcky) [inline lectures/scipy.md]:
```suggestion
The first few exercises concern pricing a European call option under the
```

**Pull request merged**